src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -7130,24 +7140,187 @@ public override IOperation VisitArgument(IArgumentOperation operation, int? capt
+
+                        int extraValueCaptureId = GetNextCaptureId(outerCaptureRegion);
+                        AddStatement(new FlowCaptureOperation(extraValueCaptureId, operation.Syntax, visitedValue));
+
+                        var extraValueCaptureRef = new FlowCaptureReferenceOperation(extraValueCaptureId, operation.Operand.Syntax,
+                            operation.Operand.Type, constantValue: operation.Operand.GetConstantValue());
+
+                        assignment = makeAssignment(property, extraValueCaptureRef, operation);
+                        initializerBuilder.Add(assignment);
+                    }
+                }
+                LeaveRegionsUpTo(outerCaptureRegion);
+
+                return new AnonymousObjectCreationOperation(initializerBuilder.ToImmutableAndFree(), semanticModel: null, operation.Syntax, operation.Type, operation.IsImplicit);
+            }
 
-            IOperation cloned = operation.CloneMethod is null
-                ? MakeInvalidOperation(visitedInstance.Type, visitedInstance)
-                : new InvocationOperation(operation.CloneMethod, visitedInstance,
-                    isVirtual: true, arguments: ImmutableArray<IArgumentOperation>.Empty,
+            // Build an operation for `<implicitReceiver>.Property = <capturedValue>`
+            SimpleAssignmentOperation makeAssignment(IPropertySymbol property, IOperation capturedValue, WithOperation operation)
+            {
+                // <implicitReceiver>
+                var implicitReceiver = new InstanceReferenceOperation(InstanceReferenceKind.ImplicitReceiver,
                     semanticModel: null, operation.Syntax, operation.Type, isImplicit: true);
 
-            return PopStackFrame(frame, HandleObjectOrCollectionInitializer(operation.Initializer, cloned));
+                // <implicitReceiver>.Property
+                var target = new PropertyReferenceOperation(property, ImmutableArray<IArgumentOperation>.Empty, implicitReceiver,
+                    semanticModel: null, operation.Syntax, property.Type, isImplicit: true);
+
+                // <implicitReceiver>.Property = <capturedValue>
+                return new SimpleAssignmentOperation(isRef: false, target, capturedValue,
+                    semanticModel: null, operation.Syntax, property.Type, constantValue: null, isImplicit: true);
+            }
+
+            static bool setsAllProperties(ImmutableArray<IOperation> initializers, IEnumerable<IPropertySymbol> properties)
+            {
+                var set = new HashSet<IPropertySymbol>(SymbolEqualityComparer.IgnoreAll);
+                foreach (var property in properties)
+                {
+                    _ = set.Add(property);
+                }
+
+                foreach (var initializer in initializers)
+                {
+                    if (initializer is ISimpleAssignmentOperation simpleAssignment &&
+                        simpleAssignment.Target is not InvalidOperation)
```

Comment:
>is not InvalidOperation [](http://example.com/codeflow?start=48&length=23)

Similar suggestion as above, consider checking for IPropertyReferenceOperation instead since we perform a cast below.